clients/rest/rest.go

```diff
@@ -123,6 +126,46 @@ func (c *Client) SetProxy(proxyUrl, user, password string) (err error) {
 	return
 }
 
+func (c *Client) SetCACerts(caFilePath string) (*Client, error) {
+	caCert, err := os.ReadFile(caFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	conf, err := c.setTlSConfig()
+	if err != nil {
+		return nil, err
+	}
+	if conf.RootCAs == nil {
+		conf.RootCAs = x509.NewCertPool()
+	}
+	conf.RootCAs.AppendCertsFromPEM(caCert)
+
+	c.setSSL(conf)
+	return c, nil
+}
+
+func (c *Client) SetTLSCerts(certs ...tls.Certificate) (*Client, error) {
+	conf, err := c.setTlSConfig()
+	if err != nil {
+		return nil, err
+	}
+	for _, v := range certs {
+		conf.Certificates = append(conf.Certificates, v)
+	}
+	c.setSSL(conf)
+	return c, nil
+}
+
+func (c *Client) setSSL(conf *tls.Config) {
```

Comment:
This and the SSlVerify function will override the *tls.Config on the client Struct.

Need to add nil check and update if present

Reply:
updated